hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/datanodes/datanodes.tsx

```diff
@@ -126,7 +126,7 @@ const COLUMNS = [
     key: 'hostname',
     isVisible: true,
     isSearchable: true,
-    sorter: (a: IDatanode, b: IDatanode) => a.hostname.localeCompare(b.hostname),
+    sorter: (a: IDatanode, b: IDatanode) => a.hostname.localeCompare(b.hostname,undefined,{numeric:true}),
```

Comment:
```suggestion
    sorter: (a: IDatanode, b: IDatanode) => a.hostname.localeCompare(b.hostname, undefined, {numeric: true}),
```

Reply:
Note to self and others reviewing this: using `undefined` here defaults to system locale.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare#numeric_sorting

Reply:
Completed space  Changes as suggested by siyao.